odl/solvers/functional/functional.py

```diff
@@ -843,45 +846,68 @@ def __init__(self, func, linear_term):
         super().__init__(space=func.domain,
                          linear=func.is_linear)
 
+        self.__functional = func
+        self.__quadratic_term = func.domain.field.element(quadratic_term)
+
+        if self.quadratic_term < 0:
+            raise TypeError('`quadratic_term` {} must be non-negative'
+                            ''.format(self.quadratic_term))
+
+        if linear_term is not None:
+            self.__linear_term = func.domain.element(linear_term)
+        else:
+            self.__linear_term = func.domain.zero()
```

Comment:
Fair enough, but there's still the issue that the functional doesn't work with e.g. `RealNumbers` like that.